web_ui/frontend/app/director/components/DirectorCardList.tsx

```diff
@@ -0,0 +1,75 @@
+import React, {ComponentType, FunctionComponent, useMemo, useState, JSX} from "react";
+import {Box, Grid, Pagination, TextField, Typography} from "@mui/material";
+import {DirectorCard, DirectorCardProps} from "./";
+import {Server} from "@/index";
+import {BooleanToggleButton, CardList} from "@/components";
+import useFuse from "@/helpers/useFuse";
+
+function searchObject<T>(o: T, search: string){
+    const objectString = JSON.stringify(o).toLowerCase()
+    return objectString.includes(search.toLowerCase())
+}
+
+interface DirectorCardListProps {
+    data: Partial<DirectorCardProps>[];
+    cardProps: Partial<DirectorCardProps>;
+}
+
+export function DirectorCardList({ data, cardProps }: DirectorCardListProps) {
+
+    const [search, setSearch] = useState<string>("")
+    const [pelicanServer, setPelicanServer] = useState<boolean | undefined>(undefined)
+    const [serverError, setServerError] = useState<boolean | undefined>(undefined)
+    const [serverDowntime, setServerDowntime] = useState<boolean | undefined>(undefined)
+
+    const searchedData = useFuse<Partial<DirectorCardProps>>(data, search)
+
+    const filteredData = useMemo(() => {
+        let filteredData = structuredClone(searchedData)
+        if (pelicanServer != undefined) {
+            filteredData = filteredData.filter((d) => isPelicanServer(d?.server) == pelicanServer)
+        }
+        if (serverError != undefined) {
+            filteredData = filteredData.filter((d) => serverHasError(d?.server) == serverError)
+        }
+        if (serverDowntime != undefined) {
+            filteredData = filteredData.filter((d) => d?.server?.filtered == serverDowntime)
+        }
+        return filteredData
+    }, [searchedData, serverError, pelicanServer, serverDowntime])
+
+    return (
+        <Box>
+            <Box sx={{pb: 1}}>
+                <TextField
+                    size={"small"}
+                    value={search}
+                    onChange={(e) => setSearch(e.target.value)}
+                    label="Search"
+                />
+                <Grid container spacing={1} pt={1}>
+                    <Grid item>
+                        <BooleanToggleButton label={"Pelican Server"} value={pelicanServer} onChange={setPelicanServer}/>
+                    </Grid>
+                    <Grid item>
+                        <BooleanToggleButton label={"Error"} value={serverError} onChange={setServerError} />
+                    </Grid>
+                    <Grid item>
+                        <BooleanToggleButton label={"Down"} value={serverDowntime} onChange={setServerDowntime} />
+                    </Grid>
+                </Grid>
+            </Box>
+            <CardList data={filteredData} Card={DirectorCard} cardProps={cardProps}/>
+        </Box>
+    )
+}
+
+const isPelicanServer = (server?: Server) => {
+    return server?.webUrl != "";
```

Comment:
With the merged change to the director response, now there's a dedicated field `FromTopology` for this.